core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -1,25 +1,32 @@
 from dataclasses import dataclass
 
 import pandas as pd
+from openpyxl import load_workbook
+from openpyxl_image_loader import SheetImageLoader
+from werkzeug.datastructures import FileStorage
 
 from core.validation.failures import ValidationFailure
 
 
-def validate(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValidationFailure"]:
+def validate(workbook: dict[str, pd.DataFrame], excel_file: FileStorage) -> list["TownsFundRoundFourValidationFailure"]:
     """Top-level Towns Fund Round 4 specific validation.
 
     Validates against context specific rules that sit outside the general validation flow.
 
     :param workbook: A dictionary where keys are sheet names and values are pandas
                      DataFrames.
+    :excel_file: the Excel file before it is converted into pandas DataFrames.
     :return: A list of ValidationFailure objects representing any validation errors
              found.
     """
-    validations = (validate_project_risks, validate_programme_risks)
+    validations = (validate_project_risks, validate_programme_risks, validate_sign_off)
 
     validation_failures = []
     for validation_func in validations:
-        failures = validation_func(workbook)
+        if validation_func == validate_sign_off:
+            failures = validation_func(excel_file)
+        else:
+            failures = validation_func(workbook)
```

Comment:
Could we instead pull `validate_sign_off` out of the for loop and call it afterwards like:

```
if failures := validate_sign_off(excel_file):
    validation_failures.extend(failures)
```

Reply:
Yes.